collectors/proc.plugin/proc_self_mountinfo.c

```diff
@@ -360,6 +360,18 @@ struct mountinfo *mountinfo_read(int do_statvfs) {
             else {
                 mi->st_dev = 0;
             }
+
+            //try to detect devices with same minor and major modes. Within these,
+            //the larger mount point is considered a bind.
+            struct mountinfo *mt;
+            for(mt = root; mt; mt = mt->next) {
+                if(unlikely(mt->major == mi->major && mt->minor == mi->minor && !(mi->flags & MOUNTINFO_IS_BIND))) {
+                    if(strlen(mi->mount_point) < strlen(mt->mount_point))
+                        mt->flags |= MOUNTINFO_IS_BIND;
+                    else
+                        mi->flags |= MOUNTINFO_IS_BIND;
+                }
+            }
```

Comment:
I mounted `/sys/fs/cgroup` to `/tmp/a`

```bash
$ mkdir /tmp/a
$ mount --bind /sys/fs/cgroup /tmp/

$ cat /proc/1/mountinfo | grep 0:26 | awk '{print $3 " " $5}'
0:26 /sys/fs/cgroup
0:26 /tmp/a
```

The current logic would mark `/sys/fs/cgroup` as bind, correct? Was the idea to replicate logic from [this stackoverflow script](https://unix.stackexchange.com/questions/18048/list-only-bind-mounts/346444#346444)?

Reply:
Hmm maybe...

The mi->root, mt->root there will give `/` `/` in what Ilya tested, and `/tmp/foo` and `/` in my example. I guess it works since in Ilya's case if they are the same will fall back just to doing the order check...